### PR TITLE
Add possibility to generate test coverage reports with jacoco

### PR DESCRIPTION
**Pull request by MishaDemianenko**
Introduce new "testCoverage" maven profile that will allow to generate
separate and aggregated test coverage and reports for each module.
Small cleanup.

### CODE REVIEW

**Review by klaren — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -686,6 +686,77 @@
...
+                            </goals>
+                        </execution>
+                        <execution>
+                            <id>merge</id>
+                            <phase>verify</phase>
+                            <goals>
+                                <goal>merge</goal>
+                            </goals>
+                            <configuration>
+                                <destFile>${project.build.directory}/jacoco-aggregated.exec</destFile>
```
Comment (lutovich) [inline pom.xml:726]: Should `jacoco-aggregated.exec` be added to `.gitignore`?

**Review by lutovich — COMMENTED** [review]
Change looks good to me. Couple questions/suggestions:
 1) should usage example and output be documented somewhere? maybe in commit message?
 2) looks like jacoco maven plugin will not work with `DsequentialTests` because it sets `forkCount` to zero. do we care about this?

File: pom.xml (hunk 1 above)
Comment (MishaDemianenko) [reply pom.xml:726]: project build directory is a target directory and those are ignored anyway already

**Comment by MishaDemianenko** [comment]
@lutovich its only first step, the final goal is to have something that will report automatically on a PR what is coverage of a new code or diff in coverages with and without PR. So people will not gonna need to do anything themselves - thats why it's not that documented sort of things.
for 2 - correct I do not think we care much since our forking model works quite well

**Pull request merged**